Input = ["""{{{{{}},{{<!!!>>,{<!>},<!>},<<a>}},{{},{}},{{{<}>}},{{{<e},!!!>,io!!!>!>},<!>},<!!!>},<h!!>},{}}},{<ee!!!><>,{<,}}<<!!,<{,!!}ie!>>}}}},{{<a!>},<!!!!!>},<!!!!!>!!!aehh!e>,<euh{!!!>!!}!,!!!>!>},<a!>},<ih!!!o>},{{<!!!>!o!>,<e!!{h{h,h!!u}a!!!!!>},<!>},<h>,<!>,<{<oai,he>},{{},{<h!>,<h!>},<!!ai<!!!!!>},<,{e!>,<>}},{{{},{<eha<!!u!!!>!u>}},{<ahie!!{eh!>,,!>>},{{<!!!!}!>},<!>,<}h{!!ohhu!h!!>,{<au!>!>,<hhoa,!>>}},{{{<!!{!>},<{u!e{i!>,<{!!ha}!>,<<!>},<>}},<!!!!<!!!>}}u{!!!>>},{<!>},<}!>},<!>h!!,u>}}}},{{{<{<h!>},<!>},<!!!>!!hoi!>},<>},{<{{!!!}}{!>,<!!!!!!!>!!!>!!!>!>},<}!!}>,{}}},{{{<h,!>,<!!>},<{{>},{{{<!!!>o{>},<u!!!>},<!>},<h!>,<!>},<,e!,h!<!!!><<ih>},{{<!>,<}o!!!!}h!!!>a!!!!!><h,h}>}}},{{{{<e!!!>!!!>!!!>!>,<!!!>},<!>!!!>>}},{<!!!!!>!>,<eaoi}{!!!>a<,>,{<ei{,o!ahee!>},<ah!<hu!uo>}}},{{<hi}!{e<!!}hu!!!>},<h}!!a>,<!>!>}}>}}}}}}},{{{{<!>},<!!!>,<}!!!>e}}i!io!!!!!>}ia>,{<>}},{{<!>o!{h!>,<!>,<u}{!>},<>},<o<u!>,<!>{a!!!>o!>,<!>},<hh>}},{{{{<!>},<!{!>h!!!!{,aihu!!!!ahiho>,<!>},<!!!>ohe!>,a<!eu}iih{>}},{{<o!>!<!!!>,<!!h}!{!>},<!>,<!!i<h!!h>}}},{{<,a!!!>e},uhu{!!!>>},<e!>}{!!u!>!!h>}},{{<!!i!,!!<}!!{!!ah>},{{<i!h!!!>h>}},{<!!h!!!>!!oi!<!!{!>{a<<!>,>}}},{{{{{<h<o!>},<{ui!!>},{}}},{<o!>>,<!>,<>}},{{{{{<!>h,!>!!!>!!!>!ia,>,<>},{<uo!!e!>,<u>}},{{<ia!>,<!a!>!>,<,ueu!><o!!<!!o!>>}}},{{{<!>},<{!!eeh!!iee>},<,i!!!>u!>},<}!>,<huo!hu!>,<!!!!h>}},{{},{{}},{<,h>}}},{{<!{!ahiuiei{{ie!{{!>,<>}}},{{{},{{{}},{{<!!!!!>u!>,<!!!>ho!!!>h{!>},<}h!>,<!>,<!!!>h>}},{{<!!!!!>!>,<<!>},<!!!!a!!i!!i!!h!>},<!!uh>,{<!!aeh!!<!},aii}u>}},{<!!<,!>},<h,!>hu!>,<!!uh>}}},{{},{<}a!!!!ioi!<!!>}}},{{},<!>,<!}oh}i!!!>e!!!>iu!!<>}},{{<}!,{!o}h!>!{!>},<!>},<a,!!h!!!>,<<>,{<e!!ho!!i!>},<!>,<}},<,hai!!!!!<uo>}},{{{{{<{uh!>},<>,{}}},{{<<aa!>},<!>,<!!!!!>},<!}e}!>},<!>,<h{!>>}},{{<!!!>},<!!!>ia!>,<{i>,{{<e,!>o>}}}}}},{<!!!>!!ai}{!>,<eha!>},<>}},{{}}}},{{{{}},{{{<oe{hooe!>,<!>},<!!h!!!>!!!!!!h!!!!!>,<}>}},{{<!!!>!!!!!>},<!!!!!>!!{{ae,!!eh>,{{<hao{>}}}},{{<iiueo>},<}}o!!i>}},{{<{i<<!!!>!!a>,{{<ee{o!!!e,{,>}}},{<!>},<}a,i{i!!!}{!<!>,<>,{<!>uh!>,<,!!!>h!!o>}},{{<e!>},<ei!>,<!!!>!>,<oi>},<ii!!!>h!uhu<!!,!>,<!h!!!}!>,<!!!>!>},<i}>}}}},{{},{<}ea!>u,!>!!}ehhh!!!>,<!!a!>},<!!!!i>}}},{{{{<!>h!!!>{!>},<hh!!!><o!!!>>}},{{{{<!!!!!>!>},<u!!!>>,{<!>!!{>}},{<h!!!>a!!!!ea!!oouh!u!>},<>,{}}},<!><!!!>u!!,!!!!!>>},{{{<!>}}!!!>},<!>,<}h!o!>,<h!>h!}o!!ihih>},{<!>ha{!,hi!!!>a}o{ehu!>aiu>,{<}<a!!<>}}},{<!>},<a!!{!!!!!>!>},<i}!!!>oi!>!ea!>,<!>,<!>},<h>}}},{{<,e}o!!<i{!!<!a!!!>!!i!>>},{}}},{{},{<a<u!>,<!!!!!{i!!e!!!>!>,<>}},{{{}},{{<<!>},<h>},{{}}}},{{{<,!!!>},<!!!,oo,>}},{<!{o!><!e>,{}},{<h!>e!!!>u!>,<{!!!>!>,<o!}eo<!!{>}}},{{{{{}}},{{{{}}},{{<!ho!o!!!>!!oeh!i>},{<!>,,ea!a!>},<e!!,h!,!!!>,<!!!>!<!!>}},{<<o!>,<h<!!!>!>},<<!!o!!h>}},{{{},<!>,<u!!h<!!!>},<!!!>ahu{!>>},{{<!!aih!>},<!o<e,,!!!!!<!!!!u!!e!>!>},<{!>,<>,{<eh<i!>},<>}},{{<<uiu!!i,h!>,<a!!o!>},<h}!!}hi!!<a>}},{<h>,{<!!oaoi{>}}},{{{{<,euia!eh>}}},{<!!!>!!}!!oha!!!>!>}io!>o<!!!>,<,i<>,<!!!!!!!a>},{<ou!!u!>ih>,<!!}}u<{,!!i!>,<}!}>}}}},{{{{{{{{},{{{<!{e{,oi>}},{{<!!!<<!>,<o>}},{<!>,<{he!>,<hu!e!>!!!>h!oa!!h!!u>,{<h!!!>a!>!>,ea!!!!,h}ii!>},<>}}}},{<>,<h>},{{{<!>,<!><hi>},{<,{e!{!o!!!>>}},{{<ih!!!>,<e!>{!>},<!{!!i,oeu<hhh>},{}},{}}},{{<h!>,<!he!!h!!!>!>},<!!,>},{<i!>!>!}<!!o!!<!!!{!h!!!>!!!>!>,<<<!!!!>,{{<h!>,<u!>i!!i!eh!!!>!>!>,<!>},<!!,h<!h>}}}},{{<,!{!>,<!!u!!!!!>ueha>},<a!hi!!!!!{aeh!!!>!!hhhho!>,<h!u>}},{{<!!h,!!>,{{<!>,<!>!>,<hh!aoa>}}},{{<i<!>},<uhou!!},!!!>!!h!!>}},{<}!>},<!>ahhe!!!>ui!>},<e}!!<!>a{>,{}}},{{<ah!>},<<!>,<o!>},<!!oh!!!!!!!>h!<!!!hi}h>,{}},{{{{<{!o!!!><,o{!>ueuoi!!!>{!!!!}!>},<>},<eoo!>!,iii!ui!>,<!>},<!>ih,,!!!>,<>},<!u{i}<!>,<u!>!>},<!>!!ai!u!!ia!!u!>},<>},{<aaou>}},{<!!!!!>},<,!!e<e!!!>i!>},<oh,o!!!>},<h>,<!<!!,!eh<u!>},<!>!!,!>},<!o>}}},{{{{{<!!!>oi>},<ui!>,<,!!o<!a},!>,<hou!>,<!h<a{>},{{{<!!!>},<e,!>e{!!{h>},{}},{<he!!h!>,<!>,<<>,<!!!>!>!}!>,<}i!>,<,!,>},{<!!!!o}{!!!>!!u{!>},<!!!!!!!!{iu!!!!!!!>,<,!!>,{}}}}}},{{{{<o!!}!!e!><!>,<!>},<}e>,<!>!!!>!>e!!hh!!!>u!!h!u>}},{{<h!>>},{<!!!!!>a!>},<!!!>e!!!>!>},<!u{!u!>},<u!>,<>}}},{{{<>},{{<!>},<!>},<!>i}!!!>,<huu!o!!{}{!!!>,<o!!!>>,<}e>}}},{{<io!!!h{!!hi!!!>heh}a!>u!!!!ui!>},<>,<ao!a!uo!>,<!!!!,i,,u!>},<>},{{<o!!!>uh!he!>,<!>,<!>},<!hu!!}!>,},e>},<a{h!>,<ha}!!!>!>,<o!!!>!a!>,<u!!i!}!>},<uh>},{{<!e!,u!>},<!>,<{a!>,<!!!!!{!u!>,<h!>,<!!h!>,<!!!<>},{<hh!>},<>}}},{{{<!!{{>}},{{<h!a,h!!!>{!>},<!>},<!>},<!!!>!!},e!e>},{{},<!>},<!!!!a!huau!!!!ho>}},{{<!!{,>},<{u!!!!!>>}}}},{{{{{<h!a!!!>!>},<oh!>o>,{{},<{!!!>!!a!!}!!e!!ai>}},{<eeuue>},{{<>,<!!!>ao!>a<,a,aie!>},<!>!>},<e!!>},{},{{<!>eoiu!>},<>,{{<{!!!>!!!!!!!>u!!!>!>},<!!!>>},<},h!>u!hao>}}}}},{<!>},<h!>},<!a!>,<h}ui!>,<<!!!!,!!<ah!!e>},{<}o!,,!>!>},<>,{}}},{{},{{{<{!!!>o!>ii!!{ou{<!!h!!!>h>},{{},{<!>},<u>}}},{{<!!!>oe!>,<>}}},{<>}},{{{<!!!ho!>,,h!>},<h{u{!!!>},<!!!>!h!!!>>},{<!!{,>}},{{<<o!!!>!!!>i!!!>h!!,!!!!{!!!>!!!!!>{!>,<o!!!>>},<<<e!!!!!!!>!!h,euhe}}!>!>},<>},{<!oa!>oa}!au!!h{!!!!!>h{!>,<}!>},<!!!!!>>}}}},{{{{},{{},{<,!!!>!>,<>,<!,,uh!>ah!>{i>},{{},{{{<!!a}i<}!>,<u!eo{!!,uih,>}},{<!!!>ae,{!>e!!!!!>>}}}},{{<{!!!!!>i!!!!!a>,{<!!!>},<!>,<!!!>}!!!>,>}},{},{{<e<{!!h{!>,<ah!>},<<!iehhh>,{}},{<!h!e}!!!>},<,oeuo!>!!!!!!!e>},{{<u!>,<!!!>>},{<!>,<u!!e!!eh>}}}}}},{{{<>},<uau>},{},{{{}},<a!>,<{ah!}!!!>!!!>heu!!!>eii<{e>}},{{{<!!hh!}i!hu!>{oh!>,<>},<a!>!!h}uh!>,<!>,<!>},<h!!!>},<<!o!!i>},{{},{{<!!!>i!!!!,{>},{<,uhhi!><>}}},{}}}},{{{{{{<u}!!<o{!u!!u>}},{<h,<!!!>}e>,{<!>ao!>iha{!>!>},<ou!!<!!,!hu}!>},<>}},{{{<ui>}},{}}},{{<!>,<!>,<a!>,<}!>},<i!>,<e>,{}},{},{<o!!,>,{<h!!i,!!!>aeh>}}}},{{{{<he!!!>hu!!!!!>},<!!i>},{<!!!>},<u{,!>},<!!!!ia{!>,<!>},<{>}}},{{}}}}}},{{{{}},{<>},{{{<!>,<e!!!>>},{<!!!!!>,>}},{}}},{{}},{{},{<ho!>,<!!!!h!!!>i!i>,{<!>,<<<>}},{{{}}}},{{{{{},<{!!e!!!!!!!!{a!!!!e<!>ie,e!>},<oh!!!!>},{<ihe>,<!!!!{!!h!>,hiu!>},<o!a!!>},{<u!<!u!>!!!>},<}!>},<!>>,<uu!,>}},{{<!>,<,!!!>!!!>!!!>},<!!!>a!>,<>},{{{<i!>!i>}},{<!!!!!>hhaa!>},<!e}!!!}o}>}}}}}},{{{{}},{{<!>},<ah!>},<<a!uhu{!!!>,<!>,<!>},<}<h>},{{<e!!h!}!>},<,!o!!oeoa>},{<u>,{{<ah<!hh}!!!>!!ehh!!>}}}}},{<{!o!>,<!>,<i!>},<i}u<io!!!!!>,<h{!>,>,{{<!!!!!!!!!>>}}}},{{}},{{{<}h!!a{!!h!>},<!!<hi!aiua!!h{>,{{<,!!e!!a!!,e!!oua<<i!!!>!o<ah!>,<>}}},{{},{<!>},<h}}ha!>,<i!!!>,<a!!!!!>hh}o>}}},{}},{{{<!!e,!!!>h!{o!>,<!!i,!>,<!!io}>,<<!>},<!>>},{<!!heo}<!>},<}!!!>},<h!hhe<!>},<,u!>},<!>},<>,{{<!>,<{!>,<!!!><{o{ieu>}}}},{{{<{!!!!!}{ha!!uuh!u!!aa!!!o!!!>!>,<h>,<!>},<!!!>}{!!h>},{{<!>},<!>,<!>!>,<u!o{>,{}},{}},{{<!!<<u!!!,ie!>},<!!!>h>},<eh!!<ah}u{{!>h!>},<a!e!>,<!>{e>}},{{{<i!>},<!!!>,<ho!!!>},<!!oa!>e!!h,eu>},{<}!!!!!>{}{ihhh!!oh},<!!!>>}},{{<<ou!!{!!h!>},<ha!>},<!>,!>},<!!!>!<!!!>},<>},<hh{a!>},<{hia!>e!>,<>},{}}},{{{<!!!>},<<o,u>,{<!>},<h!>,<hau>}},{{<!>},<i!>},<{!>,<u!o>},{<u!<,!>{u<<!!!ia>}},{{{<!!!!!>a<!!o!>},<e},}{ehu{u}>}}}},{{{<!!!>,ueiu!!>}},{{<ho!>a>},{{<iu!!>},{{{<!>,<u!!!!,,ei!>,<u>}}}}},{{{}},{{<!!!>hu<iau!>},<a!!!!oe>}}}},{}},{{<!!!>e!>},<!{{}!!!!!uho{u!!h!<{e>,{}},{<{i!!!>,<!!!>>},{<iiu>}}}},{{{{{{{<e!!!>},<!!u,i!!!e!>}!i!!!>>,{<a!>,<ah!!h!e!!!>!>,<e!>},<h!>},<!ia{!>,<h>}},{{<!!{e,!>},<ee!!!<<hh!!aao!>,<!}<!!e}>},{<!>,<!>},<,!!!>!<!>>}}},{{}},{},{{{},{<{>}},{{},<!>!>,,!>},<i}!!}!!!>{!!!>e>}}},{{{{{<,ei!>,<!!!>},<h!!>},{<,e!!!ua<a!!!!!>h!>ia!!<}{!!!>},<!h>,{}},{{<!!!>!>,<ah!>},<<!!!>he!!eh!!!>!>!>ea>,{}},<!>ho>}}},{{{<>},{<oe!<!!!!ie!!,!!!!!>!oa!>,<{}!!!>,>}},{<>},{}},{<!>},<!uu>,{}}}},{{{},{<!!!!!>u,hh!!!!!>},}}!!<{uh!,{>}},{{{{<!>i}a!>,<!,,!>oa!>,<u>}}},<u!>}{iha>}}},{{<o!!!>!>!!i{h,!!}>,{<!!,,<!!!>a!>},<}!<!o<{>,{<a!!{e!,o>}}},{<,!!i!!hio!>,<!!!>!!!><i!uiu!>},<>,{{{}},<i}i}h>}},{{<hh!!,i!!!>{}!!<uohe>}}}},{{{{<!>,<<{!>,<!!<>},<ohe!!!!!!!>o!!>},{<!,!!,!>,<!>{!>,<,o}{o!>,<!oa!ih>},{<!>},<<!>,<}e<>}},{{<hiei!i<!!!>o,!>,<hh!!!>a!!!>{!>,<e<>},<o>},{}}},{{{<!>e!>},<uiei!!>},{{{<!!{>},{<>}}}},{{{{<<i!>,<u>}}},{{{<}!!!>!>,<ah!!!>!!!!}ha!!!!!!!!!><}!>,<uo>},<!o}!!!>ah!!,!>{>},{<a!>},<u!!!>huu!!{h}h!a{!!u!!!>},<h>,{<!>,<>}}},{{<}>,<!>,<o!>h!!!>!h>}}}}}},{{{{}},{{<!ah,h>},{}},{{{{<!!ee!>,<!!ue!>,<uu>}},<!!!<uhhhuu,!!<!>},<!>o!>>},{}}},{{},{<>},{{<!!{o<!!!!!}auh>},<!!!,,<!!uh!>},<u!>h!!!>h<<!!!!ui!>,<>}}},{{{<h,oh}!>,<}!!!>!au>},{{<o!<!!!>,<ie{!!{heu!!!>!>i{>},<!>},<!!!!!>!>h<!>},<hie,!>!!!>>}},{{{},{{{<!!he!aeh!>!!!>oo!>,<h>},<!>h{<>}},{<h>,{{<ahe>}}}},{<{!!,>},{<{!!}!>},<!!hii{!h!!!><!>,<!>,<!!!!!>>,<!!!>!<ee}u!!!i!uei>}},{{}},{{{{<!!!i}h!h>,{<<{uhh!!a!>,<>}},{{<!u,!,{he<oo!>>}}},{{<oo!!u!!ehh!i!>>},{{<ai!>,<h!>},<{!>}!>},<>},<!>,<a<>}},{<i!>},<!>},<!!!>},<,<!h{h>,{<h!eh!>,<!!!>!>,<>}}},{{{<!>,<!!>}},{}},{{<!!!!!>i!>,<i,u!>},<u!!e!>,<hh!>},<{h!>!>},<>},<!h{!h<<!>,<{!{ho>}}}}},{{{{{<!>!>a!!}i!>,<!!u!!{!>},<>},{}},{{<h!>h{!!h!!{!!!>!!h!<!<!!!!!>},<>,{{<i!u!>,<h>}}}},{{<,<a!u!!!>,!<i!>,<!i,>},<!>o!!o!u>}},{{{<h,h!!!>,<!>,<h>},{{{<>}},{}},{{{{<!!<!!,oi>},<!!<>},{{{}},{<!i!h}a!>},<oha>,{<h<he!!!!{>}}}},{<!>},<!>},<}!i!!h,,!!!>}!h!!!>,<!!!>>,<!>,<eu!>},<!o!!!,<<<!!!>>},{{{{},<,u!>},<i!!!!!>},<e>}},{{{}},{{<!>,<!!!><!e!{u!!e{<<,}!>,<{!>!!}!!h>}}},{{{{<!>aa!!!>}!!!>o,h!>},<}!!}!ie!>,<h!!!>i!!>}},{<h!>ho!>},<!h}a!!!>!>,<!he<!!{!>,<!!}h>}}}}}},{},{{{{{{<!!!!h>}},{<!>,<{i!>,<!>},<ohoi!h}>}},{{<!!!><{u>},<!>},<!!!>!!!h!!ihu}!u>},{{}}},{{<o<,!!!>ou!!!>,<uee{h!><<h>},{<e<o<,,!!!!!!<<>}}},{}},{{}}},{{{<!!a!!!!!!,a}!!h!,hoi!>,<o>},{}},{<{!!!!!>},<!!h!hh!!!>!i!!,!>},<!>},<o!>},<,ia>,{<!>,<>,{<eu!!,!!!>a!!!!!>!!a!>,<}h!!!!{h!!!>>}}},{<!!!>!>,<>}},{{},{{{}},{{<,!<a!!}{<{>}}}}},{{{{{<>}},{{<!>eh!!!><!!ah<!!,!{o!ua!>},<i>},<!!!>u!!!!!>!!!>{!>o!eae}!!!!o!!o<!>,<h!>},<>},{}},{},{{},{{{<e!!u{o!>!!!!!>,<e!>!>!!!>h!u>}}}},{{{{<!!!>oe!!!!!>!!!>!o>},<!>!!h!!i!!<e!>!>!!!>a<!!h<!>,<i,u!!e>},{{{<!!!>,<ah!!!!{huu!!!>}u!>>,{<h!>,<oeo!!,!>,<,a!!!!{<u!!!>u!uo!>>}}},{<!>i!>,<{!!!>,<{!!a!>>}},{{<hia!e!!!!!!!>ua!>},<>},<!!!>!!,!>i>}},{{{{<!>!!{,>}},{<>,{<oh,>}},{{<>,<h!>,<uia!>!!h!>},<}h!hh!>,<h}>}}},{<!!!>{{!h!>,<!!!!hi!!!>!>,<a>,{<i!>},<}!!u{!e!>>}}},{{<!!!>!!o{,oa>,<hu!>!{!!!!!>!>,<!!a!<ae!!!>!!!!hu<{!!!>>}},{{{<>,<!!>},{<,!>,<!!!!!>!>},<,h>}},{{<!!a}}!>},<oh,,i{uih!!!>}o,u>}},{}}}},{{{<!!!!!ihh!>},<<!h!!!>,h!!!>!!i!!,!!!!u>,{}}},{{<h!!!>}!>,<!!!>!!<>,{<eoh>,{{{{<}!!o!>,<{u!!!>>}},{<!!}hi!>},<h!!!>,<h,!>h!!!>!!!!!>h!ie!>!!{>}},<u!>},<}!!eo!!!!!e{!!!!,!>,<!!h{!>>}}}},{{<i>,{}}}}},{{{{<!!!>,<!>,<h!{>},{}}},{{{{<iio,!>,<a!>},<o!!,,a!!!!!>>},<!!!>i!<{!a!!ui!!!,<a!>>},{<!!h,>,<>}}}},{{{{}},{<!!!>a{!!!>}u!!!>},<!>},<hhie<!a{>,<a!!!>!>},<h}o!>!>},<!>ha,!!!><!>iheh>}},{{{<!>ha!>,<oh!e!>,<}!u!!,!!h!>},<>}},{<u,!>!!!>h,a!!h}>,{}}}},{{{{{{},{{<ho>},{{<!o!>!!i!!!>!>,<!!!>!!u}i>},{<!!h!!!he!>},<o!!!>ae!{!oi!}>}},{{{<!!!>},<!!e!uo{!>},<!!h}!aa<!>!>},<!!!>>},<!>,<e{h!>,<o>}}},{{{{{{<i!>},<!!!><!>,<{!!ou!!!eha!>e!>},<<<!!e>}},{{<!!!>>}}},{}},{{{{{}},{<<uh!!!}}i>},{{},{{<<eh!!}!!!>}h<!!{!},>},{{{<!!!>},<,!!!>!a,o>},<aoh!!{e!}}}h!>,<!>,<,ehh!}!h>},{{{}}}}}}},{{{},<e!>!>,<,u!>},<i!>!!!>,<<u!>,<!>,<he<>}},{<},!i!>,<hu!>},<i!!!>,<!!!!!>!!!>!!},>}},{{<!>!>,<!!!i,u!!i!!!!<u!>,!>}o!!>,{}},{{{},<!>},<!!iu}o{!<!!,uh!>,<!!h}o,!>},<<h>},<!ah!!!!}!u}!>!>},<>}},{<o{,u!>},<!!i!o!!!>h{}!!!!,!o!>},<o},>,<a!{!>,<!!e!>!!{u>}},{{<h,!>,<a>},<}!>!!!>!!!>}e!!}!>},<o!!!>uhh>}},{{{{{{},{{{}},{<!>o!>!!!>,o!!!>!!,e!!!!!>i!>,<,!>},<,!>,<i>}},{{}}},{},{<i,>,{<ea!!<>}}},{{<!!{i!>,<>},{{{<u!!!>h!>},<!!<!>},<ou}!u,!>},<}}!!!>},<}a>},<h}{,a{>}},{{<<!e!}!>,<!o!u!!!!!>,<!>,<!>h!>,<!>},<!!!>o!o!>{>},{<>}}},{{<<h!>,<!<!!h!!!>u}{a!!!>!>h!!!,o>}}},{<o}<!>,<!>},<<!>},<o!!!>!!!>{uu{i!!!!!>!>,<h>,{<,ah,h>}}},{{{{{<u!>,<}!<!!hu!i>},{<!!}}!!!>,<h!!h>}},{}},{{<!!!>o!!!>hu!>},<>,{<u!!,!u,i>}},{<!!!>{,eh!>,<uiie<<!!eo>}},{{<!>,<{<uo,h>},{<}}}>}}},{{{<ioi,,}{!>>},<h!!!!h!!hi!!!>,<!>,<a!>!>},<>},{{{<<!>},<!!<!!i,}!>,<!>,<h!!!>>,{}}}},{{<!>,<!!!!!>!>,!!!>,<>},{<o,uhueoo!!<!>},<!>!>},<!!!!!>!>,<h>}}},{{{<hh<{{!>,<!!>},{{{{},<>}},{{<!uu!o!>>}}}}}},{{{<h!>,<!>!i!heuu!>o!!!!,!{!!!!{!!!>!>},<!>,<}!>},<>},{<>,{<u!!!>!>},<!>},<}<u!>},<!>>}},{{<h!!!>,<}<<!e{i,!!!>!>,<!!!>!!!i!!!>,<!!{a!h{>},{<oh<!>,<e!!>}}},{},{{{<{{!!}hu!a{>},<!{<!>!!!!!>ihh!!!>,<uu,}i!!!>>},{{},{{<!!!>h!!{>},<}!>!!{uh!!uh!!h!!u!!{,>},{{{<u>}},<!!!<!!!!<ho!!!>},<e>}},{{{<{{o!>,<ihe!>{!!u!>},<!o,>}}}}},{}},{{<!!!>!!!>hoh!!uie>}}}},{{{{},{{<hh!!{!>,<!>},<!!}}>},<!>,<h!!,!!oi!!!>ieaa!!!>},<{h<!h!!u>}},{{<uea!>,<>},{}},{{},{{{}},{{{{},<!>},<!h,>}}}},{{<!!o,!>},<uiih<>}}}},{{{{<!!!!!>h<!!!>},<!>,<ohh,!!<>}}},{{{},{{{<o{}!!!>!o!>,<h{,>}},<!>,<!>e}u!!>},{<!>},<h<}!>},<ieu!>},<<,!>eh!!e{h>}},{{<!!!!<!>,!!hoe!!<o<!!!>{!>,<hi!>},<,i}{>,<u!>,<hi!e!!!!!!!>},<o}h!>},<a>},{<!!!>a!>,<!!auuu!!a>,<!>,<!!!>a!>,<!!!>{e}<ee!>},<{>},{{}}},{{{<!>},<o!!ou!>},<!!hhh!!<<!a!!<}!!!>!!>},<i<,i}!eh!!}!!u!!<o>},{<<o!>!!,}<e!!!><>,{<{!>},<o,uuua>}},{{<,i!!!>}!!{!!a>},<,!!!>,<!!i}!eh},!>h!!!!!>!i!>,<!!!>!>,<o>}}},{{{<oua!}!!h{<!h!oa!!!>ia!>}!>,<>,<!>},<!>},<,!!u{i!!!>!>},<>},{{<h>}}},{<!>,,!!{!!!>!>!!!>!e{!!u!!{!!!>},<{}h!>!>ao>,<!hhuu!{!!!>},<!>!uhh!>},<}h!!oi>}}},{{<h!>,<,!>!!!>u>,{<>}},{<,!>h!!<!>!>{hau!!!>!ii!{!!!>}{oe>},{<!!u!!,uo!<!!,!!!}{}a!!ia>}},{{<>},{{},{}},{{{},{{{<>},{<o,uh!>!>,}ihh}eo>}},{<ih!!!>o!!!>,<au!>heo}e!!!!!>>}},{{<!!a!!hu!!!!!!!>!>{e}!>},<{!!<o>},{<}!>,<hu!!hh{!!!>>}}}}}},{{<}!oo!>},<!i!!h}ih!!!>},<!>},<>},{{<>}}}},{{{{{}},{{<{}!!!!!>!!h<h!u}!!!>u}heh,!!u!>!>},<!>,<>}}},{{<!>,<{io!>a!!h}}{i>},{<<h>}},{<!>,<!!{!!{!>},<>,{}}}}},{{{},{{<!!}u!>!!u{!o,>,{}},{{<{oa!>h!>,<!!,!>,<>,{{<ah!!<}!!!>!h!!u!}!>},<i!{oio!!!>>},{<!>},<ih>}}},{{<ahe!!!,}i!!h{u,!!!>hi!>},<>}}},{{<!>,<e!!>},{{<!>,<{,!!i!>,<,!{!>h,{,!!ii,!>},<>},{<!>},<!<<i!>},<!>!>},<h!eai!>,<!!!>,<ha<}<o>}}}},{{{{<h!e>},<<a!he>},{},{{{<!>,<>}},{<,h!u!!e!>},<!>,<<ouh!>},<o>,{}},{}}},{{<,!!!h!>o!!!,>},{{<oh<!!a,,i!!!>!>},<!>,<!>o!>},<o{!!!>},<a}!>,<>},{{{{<h!!!>hu!!h!!a!!!!<{>},{<!>,<!>,<uue!!!>},<{h{oh!>,<!!!>!!{!>,<>}},{}},<uh!>!!u!>},<<<h!>o!>,<uh>},{{<!>>,<u,!>,<>},{<aih!!!>!>,<i,a!!auu}!>},<,o!!!>},<u>,<<{o!!!!hah,h!ie!!i!>!h!{!>>}}}},{{<{!!!>>,<,uh!oi,!>,<,ao!!!hhhho!>>}}},{{<!!eh!!!>aheu!!eee!>},<!!a!!!!,!!i!!!>!!e!!!>},<{>,<u!>,<<e!>,<,!!h!u!oaa!,!>},<a}!!!!}}>}}},{{{{{}},{<!!!>},<!>,<!!!>>}},{{{<<<a!>}haoh!>,<i!o,}!!!>,<!!a,>},{}},{{<!!!!u<}!!!>>},<h!!!>},<>}},{{{}}},{{}}},{{{{<!{a!!i}!>},<{e!>},<i>}},{{<!!!>h<u!>,<!!!><>},<>}},{{<h!!!>}!!!>>},{{},{{<!,<i!!huu!e}o>}}},{{},{<!!u!>},<!!e<u,!!!>!e!>!>!!!>ha,!>},<!>!!>}}},{{{{<,!!!>a!!a!>e!!hu!>,<!!!!{>}},{{<a!a!!hh!!eheu!o<!<!h!!oa!!!!!>{!>,<>}}},{{{<eh!!a!!!i!o!}u<!!!>aae>},{}},{<o<!>},<ue!>!oh!>,<aio>,<!!!>!!,ue!>,<!!!>},<o<!!<!!!!!>!>},<uu>},{{<u,{eh{>}}}},{{{{{<<h!a,<o>},{}}},{{<!!!<h,!!e!h!>!}!!h{}auh}!>},<!!!i>},{{{{},{<{!!!!!>!>},<!a!>>}},<!>,<>}},{<{e!!u!!eh!>!!!u!!e!>},<!!!>!>!!<!,>}},{{{{<!!u!>{h!>h!>!o!!e>},<!>i>},{{<!{!!!>,<{o>},{{{<h>}},{{<,{<a<aa!!!>ai!>>}}}},{<!>!o!u>,{}}},{{{{{<ui!>},<!!iu!>,<>}},<{!!e{a,<!!!><!!!!<o}ii!!!>},<!!!!!>oa>},{<!a<!uio!!!>ii!>},<{h!>}!!!>>,{<!!u!<!u!!e!!!>,<!!a,!!!>},<!!!!o!>},<,!>,<a{<>}},{{<!>!!euu!!ui!!h!>,<i!!hae}h!!h!h>}}},{{<h!>oo!!!>},<!!!!o}!!i>},{}},{{<!!eh>}}}}},{{{<<!!!>ui!h!>,<e!!!!!{u<!!!>>}},{{{<e{o!!!!!><e,ou{}!!!ha<},>,{{<}!>,<{o!!!>eh<!ih!!!>!!!>,<!>h!>,<h!!!>!h>},{<>,{<!!{>}}}},{{},<!>,<!>},<!}>},{{{},<!>!>,<!!ai,uia!,{!!!>}!i!a!oh!!i>}}}},{{{}},{<!>e{}!>,<!!<{<h!!e,e{{!>},<!!!>},<!!!>,<e>,<h!!,!h!>,<,}ao>},{{{<!>a}!>o!!!>,h{!!u!>!>!!{h!>},<!!!>>,{<!>},<}!>,<uo!!h!!!>e,!!}!>ih!!}u!!oa!!hu>}}},{{},{<!!<!!!>},<!>,<!!!h{,!!h!>ha,hiie>,{}}},{{<!!!<!!hho!!!>i!>},<!!,!>h!!hhe>}}}}},{{},{<io!>,<!!!>ai!!}h{>}}}},{{{{}},{{<!>,<!>,<i!!,!}!!>,<!>},<o>},{}},{{{<{io!!!><u}!o!>,!!!>},<!u<!!!><!!!>o>,<>},{<{h!!!>,<uu}!!!!!>>,{<i!!,!!!>},<!}!{!>},<!>o}!!<!>,<ao!>,<!>,<>}},{{},{<h{oa!,a,hu>}}},{<e!!!!e<uh>}},{{<<!>!>,<}h}a!!!>,ao}!h!>,<ue>,<hio!!!>,<o}>},{<<!>i>},{<<!!!!h!>},<!{ahu>,<!!!>}!>},<>}}},{{{{<!!ehu<!!!>},<!,a!!!{,!!!!>,<e!!!>!>},<,>},{<ae!o,h!!!>!!!>!!,!>},<e!!!>!!!>u,>},{{<!>},<o!>!>,e!>!>,<!>,<}!!ee!>a>},{}}},{{<<!>,<o!!!>!!u>},<hu!>},<!!!>},<ohi<ah!>},<o!>h>},{<!!!!!>,<ou}!>,<{,u!!!>,<!!!!a!i>}},{{{<!!{!>i!!!>!>,<{ai<!>},<eei!!!e>},{<o!>},<!!ehi!>},<!i<>}},{{},{{<!!uo!,hho!>},<}>},{<uh}!>},<}{!>},<!>},<}!>,<!!!a>}}},{},{{{<a!>},<h!>},<!!!>!!oh!>u!!hiau!!!!!!>}}}},{{<u!!h<!>},<a>}},{}},{{{{{<e!!!><!>i!!!!!>u!e!>},<!!!>,!uh,h{>,<!!!!!!a!!a!!<}!uo!>,<!!!!o!!!>,!!!!!>>},{{<hhi!!<!eae!>},<h>}},{<!>},<a{h<!>,<o>}}},{<!!{{eo,!>,<h!{>,<h!>!>,<!>ha}!eio}h}!>},<<{!!>},{{<huu<>},{<e>}}},{},{{{<!>},<uu!>!e!!!>au>,{{<,<},}!>},<,e!<u!!!>a<!h!>{!>},<!!!>>}}},{{{<,<>},{<!!!>},<<{o!!!!iu!io!!!!!>!>!}u!!!>!!!ei>}},{{<!!!>}!!!!<,h!>},<u>}},{}}}}}},{{{{{{<!>,<e!!a}!>!>,<e>},{<<}!>,<!!{e}ao!!!!!}>}},<h!>},<!!!>e<>},{<e!!!!!>!>},<!>!!ahh!>},<i!!!!hi!><!>},<i!!!!!!}!>},<!>,<}>},{<h!!!!,o!!!!!!!!!u!>},<,!!>,{{<ii,!!i!>},<{>},{{<ou!!{i!!<>}}}}},{{{<ue,!!!>h!>},<ho!>},<o,u>},<!!!>,<}i!!!>},<oh}h!!!>a>},{},{{{}}}},{{{{<!h}o{!!!>!!!>u{!!uae>},{}},{},{{},<!!u<!!!>!>},<!!!!!>!>,<,!>},<e<u,!!>}},{}},{{{{<ieh<!!}<>},{{<{u{!>},<!h>},{<{h>}},{{},{}}}},{{{<h!>},<h!>},<u,}!>{!}!>},<i,}<}!!}>}}}}},{{{{{<!>e<!>},<<!>},<<!>},<>},{<i!>,<o!!i!!>}},{{},{<>}},{<>,<{!!!>!,<,!!u!!{!>,<oh!!!!h,h,!!!>o>}}},{{{{{<e!>},<a!{,!>,<e!>},<!>a!>,<!!!>o!!!>,!!!>>},{}},{{{<e!!!>,<iih!!!>,<o{u>},{}}},{}},{{<e!>,<>}},{}},{{},{<,!u<}u!!<{!ea!>},<!o!o>,{}}}},{{<!!!,a>,<!!!!i<>},{<}e!o!!ei,!!!>h!!!>}}<!>,<o>,<!>,<!!!!!>!>},<i!h!<u!ee!>,<e!<h!>,<!!o!!!>},<>}},{{},{{{<>},{{},{}}}}}},{{{},{{<!o!!!}o<h,!>!>},<o!!!>},<>,<o!!}h!!<}a!>,<!>},<i{<hh>}},{<!!o,h!>!!!>}{!>},<!>,<!>!!}{<!!>,<u<!!!h!>},<ohuo!!!>>}},{{{{{{<h!>},<a!oo!>,a!!i!!!>uu>}},{{<a!>}!>},<!>h!!{eoaaa!!h{!!!a,e>},{{},{{}},{{{<{!>!}hah!>},<!>,<!>},<u{>}}}},{{},{{}}}},{<>,{{}}}},{{{{},{{{<<!>},<!>!!!>!>,<!!i!>!>!h{{e!!>},<oi!>},<{!!!>,<a!>},<!>}!!!!ua!!}!!e!>!!!e!>},<!o>},{<!>},<!>u!!>,{<!>},<o!i}ohh{!u>}}},{{{<!!ai!>,<i>}},{}}},{{<o!>},<<!>i!!{}hi!>!!>}}},{{<!>,<e!>},<!>,<!!!>>},{<!>>}},{}},{{{{{<h!>,<!!!>},<!!,!>,<<iha}!>,<!>,<o}u!!o!!!>},<!!!>>},{<{!{!!!>}!!<i!i!>,<hheu!!ha>}}}},<!>},<!>},<h<!>},<oouh!!iih!e!!!>h>}},{{<!!e!>,<o!>!!!>{h!>},<a<!!!>!!u!!!>!!!>!!u{i>}},{{{<{<!>},<u!h!>},<i!!!>,<!!!!!>,<}eh<!>},<>}},{<!!!>},<h!!!>}h}he>,{<!h{>}},{{<{>},{}}}}},{{<io!>},<i!{hhhi{,}!!!!!>!!!>!!!i<i>,{}}},{{{{<ei!!i!!>,{}},{<}!!!!i!!!>},<!!ei>},{{{{<{<,!ia!!!>h!>},<!>e}e!!!h>,{<!uh!>,<!>,<iu!!!>h!!h<<!>,<!>,<h!>},<hhi!<>}},<!!{io!!!>{!!!>!!!!!!i!!!!,u!>>}},{{}},{{<!>!!!>u!!!>!>},<<ih!!oh!>a>,<<oe<}!h!>},<<}!>},<>},{{},<o!!!ohh>}}}}},{},{{<,!<!>},<!>!!h!<!!!!!haoa{!>>}}}},{{{{{{<o!>},<>},<u!!ohi!!u!!!>i!uh!>u>},{<!>,<!!!>e{!>},<a>,{<}!>,<u>,{<!>},<>}}}},{{{},<<e!!!>!!!>},<!>h>},{{{<!>!a}{{!!ua!!!>,<hiu>}},{{<e!o!>!!!a!>,<!!!>h!!!>!huh{>}},{{<o!{uh>,<},u}a!>,<!!!h,o!>!!!>!>o{<>}}},{{<!>,<!u!>,!!!h!>!!!!!>!!!>!>!!!a!!!!!>!>},<,>},<!!,!>o!!!>huu!>e!!!!!>!>},<i>}}},{{{},{<!!!!<{!}{h!!!>e!!!>{o!>h!!u!}!>>}},{{<<!!a!>aui>},<,a!!heh,!!!}>},{{<!o{!!!!!>{!!!>oh>},{{<,h{!!,eh}!>,<e!>,<!!e>},{<h!>,<,!>},<u!>>},{}},{{}}}},{{{<!>he{!!!>!!!>ih!>},<i!>,<>},{<u!>},<!>!>},<ah!!h!>},<>}},{{<<hu!>!><}!!!>{h!>},<hheh>}},{{},{{<{e<>},{<,uu!!!>h!>,<!,a>}}}}},{{<u,h!>,<!!!>eoe!!!>!>>}},{{{<h!>},<!!!>!!!>}!!eh}>},<o}}!!e!>i!i>},{{{<!>,<!>},<!!!!}>},{<!a,!{!>,<!>,<h!!i<>}},<e!>,<iuh!>,<}e!!!iao>},{}},{{{{{{},{{<!!<!>,<!!}hooh!i<i,!!!>>}}},{{<{>},{{{<!>,<oh>},{<i!!!>!i!o!!ehe!>e,!>i!ah!>u!!!>>}}}}}}},{{<}!!!>o,!!!>!>a>,{<<a,,{>,<,!!!>}h!>},<!!!>!>!u!!!><!!h!!>}}},{{<o!!!>!<!!e!!ah!!uai!>ahh!!!>e!>},<{>,{<ee!!,!>,<<>}},{<{a!>,<,i!>!!i!u},!!!>!!!!u}h>,{<>}},{{{{<!>},<!!hh!>},<,,!>},<!!i!>,<!!h!>},<}!!h>},<!>},<ee!>},<,!>,<{!h!>,<u!!i!}>},{{<i!>},<i,!!!><ohu!>,!>},<!>},<o<!>ai!>>}},{}},{{{},<{o!>,<a!!<oe!!!>a>}},{<h!!!>!!!>>,{<{!!u,<}!>,<ho!}{iaha!>,<eu>}}}}}}},{}},{{{{<!>},<!{!!!>o!!!!!!{!>},<,}h}ho!uo!!uh!!e>},{{{{<u!>,<!>!>{u!<hh>,<i,,!,u!!!>i!!{!>,<!!a!!!>,<!>},<{h,>}},{<h,<!,!>}>}},{},{{{<<!h!!!>>},<h!>,<!>},<!!,h!>,<!!e!!!>!!<i,>},{{<!>},<!>,<ao}oe<>},<!!e!>,<o{!>,<}!!!>},<i!>},<u!!a!>,<!u}>},{}}}},{{<!>!>},<h!!h!!,!!o{}he!!o!!!>!!!>hu>,<!>!he}i!!!>!!!>!!ah,>}},{{<!!h!!!>{!>}hu!>,<<!!!!!>}e!!>},{<,u!!u!!!>!!<!>},<!!!>,<!!e{h,!!i!!!>!>},<!!o!hh>}}},{{{<!>,<}hh!>!>,<o<uih>}},{{}}},{{<!i!!!iu!>},<i!ah,>},{<>,{}},{{<{!>},<e!!!>!!!>{ua!!!>!i!au!!!oi!!!>>}}}},{{{{},<}hi!!ae!!aiu!!e>}},{{<!!{!!{i!>!>,<!>,<e{}!!h>}},{{{<!!{!!u!>},<!,!i!!!>!{a<{<!>,a>},{<e!!!><,!!!><<!!<!<o>}},{{<!>{},<!!!uh!>,<!>h!>,<!!!>{{{i>},{{<!>h!!!!>}}}},{{{<}hho!h<!!!>>}},{},{<>,{{}}}}}},{{{{{{<!!!>!!o!!!>hh>}},{{{<!!o!>,!>},<,i!!o!uooo!!,!!!!>}},<ai!>!>,<a!e{!>,<h!>,<a}ha>},{{{<!!}aoao}i!>h>},{<h!!}>}},{},{}}},{{}}},{{<!>},<!!i>},{}},{{<}!>!!!!!>}!!e}o}!!!>ue!!!>!>},<}!>},<,!>,<!>},<>,{{}}}},{{{{{<!>,<!!!!!!uu{!!!>ehui!>},<<i!hu<>},{<!>i!!!!!!{!>o!!e,!!o!a!>},<}<!!!>},<!>},<!!!>,<>,<!>,<!!!!!>!}hua!<!!!>e!!o,!>},<>}},{{<!!!!!>e}>},{<o>}}},{{{<,!>,<}}>}},{{{}},{{<,<ai!!!>},<e{>},<!!hh!!u!!!>u!!!>a>},{{{<>}},{<,oi,!!,!!}!!!>!!,e!!!>,<i!>}{io}i!!u>}}},{{},{{},{{{{},{<!>,<!!h>}},{{<a!!e{,!>!>!>,<{e{a!>a>}}},{{{<o!!!!o<!>},<a!!!>!>ha!}!o!!!!}h,>},<{!!!>>}},{<{!>},<!!!!{!!o!>,<!>,ao<>,{<{a!>},<!>},<!>}o>}}}},{<e}!>,<!!!>,<o{i!e!!!>!!}u!!!>!!}o<!!!><{>,<i!>,<{i!!}!{!>,<!!!>!>},<>}}},{{{{<!!!u,,,a!e!!>},{<!>o>}},<!!!>ha!!ohh!{}u!!i!>},<!!!>!!!>!!!>!>hh!!>},{<!>,h!>u{!>,<<!!!>!>oi!>},<e}h>,<i}!!h,,<!!o>}}},{{<!!!!>,{}},{{<>}},{{<o!!!{!!!>,{!!i}!!oh!!ah{,>},<hu>}},{{{{<!!u!!!>>}},{{},{<!!{!>},<,a!o!!!>},<!>,<>}},{<!!!<iu!!{u!>{i!>},<!{e>,{<a!!!<!!o<e!>hhh!!!>h}!!a{>,{<!!ea!!!>iah!!!>!!!!!!!>ah>,<o<{{!>,<ea{i!ih!!oh!!e!h!!u>}}}},{},{{{{<{{<h!!{!,h!>},<,a!!!>,<>},<hh!>},<o,}!>},<iua>},{<u!!hh!!!!!>!>}}{u!!!>{!>},<!!<!!!>>}},{{<h}ua!!h}}!>>},{}},{{},{<>}}}}}},{}},{{{{<>},<>},{{<!>>},{<!>,<<!!!>}!!!>ii<!>!!!>!!!>!!u!!!>!!h>}},{{<i!>},<>}}},{}}},{{{{},{{<!!eea{>}}},{{<!!{<!i<}hih!>},<>}}}}}},{{{{<!>},<o!!!>o!!<!>}i!hh!e>,{<!>,<o<>}},{},{{}}},{{{<ii!!!i}!>},<!!i>}},{{},{<<!oi,!!!>!!{<!>,!><h!!!>a>}}},{{{<!!!>ae!!!!!>},<oi!ue!>},<!!e}!>!!!>!>!}>}},{<i!>,<,!u!}>,<!!{ehho!>},<!!a!>ou!!!>!!i,!!!>>},{{<!!!>,<oo>}}},{{{<u}u<{!>,<!!!!!>!>},<{>}},{{<!!!>!e>,{}},{<i!>,<a!>,<!>,<i!>,<!!!e!>{<hh!!u{!>},<,>,{<ohae!!e!!e!!!>{i!>>}}},{<h!>!!ho!!h,{,u!!!!!>!>!>i<<>,{<!>{a!!!>!euh}h!}!>,<}!>},<o!o>}}}},{{{<!>},<!>o<!e!>},<u!>},<{<!i!>,<}e>,{<u}i!!!!i{,o!>},<u!>>}},{<u!!!>,!!!>h},!>u}<}!>,<!!!!!>!!!>}h!!u!>,<>},{}},{{{}}},{{{<i!!!a!a!>},<u!>},<!>!>},<!>},<h!{!>!!!>!oo}!!!!!>u!>>},<!!!>o,>}}},{{{{<,uh!>},<{h!!{!!!>},<h!>},<!!!>!!,!>,<!!{!!h>}},{{{<,,,h!>!!h{i!!!!!!}>},<{!!!!!!!>e!>,<u!h<!>!>>},<,{>}},{{{}},{{<!>,<,!!,o!!,a!!!u>,{<!>!!!>!>},<h!!i>}},{<>}},{<u{!>},<>}},{{{{<he!>!ih>},<!>h!>},<ue!>},<aa!>,<,!><!>},<o!>>},{{<!h!!<!>}>},{<a>}}}}}}}"""]

Input = Input[0]

def FindItems(String):
  if String[0] == "<":
    Index = 0
    Output = 0
    while True:
      Index += 1
      if Index >= len(String):
        break
      if String[Index] == ">":
        break
      if String[Index] == "!":
        Index += 1
        continue
      Output += 1
    return(Output)

  #{} type item
  String = String[1:-1]
  Index = -1
  Items = []
  Indexes = []
  Opened = []
  Garbage = False

  while True:
    Index += 1
    if Index >= len(String):
      break

    if len(Opened) == 0 or Opened == ["<"]:
      if String[Index] == "<":
        if Garbage:
          continue
        Indexes.append(Index)
        Opened.append("<")
        Garbage = True
        continue
      elif String[Index] == ">":
        Garbage = False
        Items.append(String[Indexes[-1]:Index + 1])
        Indexes = Indexes[:-1]
        Opened = Opened[:-1]
        continue
      elif String[Index] == "!":
        Index += 1
        continue
      elif Garbage:
        continue

    elif String[Index] == "{":
      Opened.append("{")
      if len(Opened) == 1:
        Indexes.append(Index)
      continue

    elif String[Index] == "}":
      Opened = Opened[:-1]
      if len(Opened) == 0:
        Items.append(String[Indexes[-1]:Index + 1])
        Indexes = Indexes[:-1]
      continue

  if len(Indexes) != 0:
    Items.append(String[Indexes[-1]:])

  Output = 0
  # print(Items)
  for x in Items:
    Output += FindItems(x)
  return(Output)
      





assert FindItems("<>") == 0
assert FindItems("{<random characters>}") == 17
assert FindItems("{<{!>}>}") == 2
assert FindItems("{<<<<>}") == 3
assert FindItems("{<!!>}") == 0
assert FindItems("{<!!!>>}") == 0
assert FindItems("{<{odi!a,<{i<a>}") == 10
assert FindItems("{{<{odi!a,<{i<a>, <random characters>}}") == 27

print("Test cases done.")

print(FindItems(Input))
